docs/guides/text/chat/chat_tools.cs

```diff
@@ -0,0 +1,68 @@
+// SAMPLE: Generate text from a simple prompt
+// GUIDANCE: Instructions to run this code: https://aka.ms/oai/net/start 
+#:package OpenAI@2.2.*-*
+#:property PublishAot=false
+
+using System.Net.NetworkInformation;
+using OpenAI.Chat; 
+
+string key = Environment.GetEnvironmentVariable("OPENAI_API_KEY")!;
+ChatClient client = new("gpt-4.1", key);
+
+ChatCompletionOptions options = new();
+options.Tools.Add(ChatTool.CreateFunctionTool(
+    functionName: nameof(MyTools.GetCurrentTime),
+    functionDescription: "Get the current time in HH:mm format",
+    functionParameters: BinaryData.FromObjectAsJson(new
+    {
+        type = "object",
+        properties = new 
+        {
+            utc = new
+            {
+                type = "boolean",
+                description = "If true, return the time in UTC. If false, return the local time."
+            }
+        },
+        required = new object[0]
+    })
+));
+
+List<ChatMessage> messages = [
+    ChatMessage.CreateUserMessage("what is the current local time?")
+];
+
+while (true)
+{
+    complete:
+    ChatCompletion completion = client.CompleteChat(messages, options);
+    messages.AddRange(ChatMessage.CreateAssistantMessage(completion));
+    switch(completion.FinishReason)
+    {
+        case ChatFinishReason.ToolCalls:
+            Console.WriteLine($"{completion.ToolCalls.Count} tool call[s] detected.");
+            foreach (ChatToolCall toolCall in completion.ToolCalls)
+            {
+                if (toolCall.FunctionName == nameof(MyTools.GetCurrentTime))
+                {
+                    string result = MyTools.GetCurrentTime();
+                    messages.Add(ChatMessage.CreateToolMessage(toolCall.Id, result));
+                }
+                else
+                {
+                    Console.WriteLine($"Unknown tool call: {toolCall.FunctionName}");
+                }
+            }
+            goto complete;
```

Comment:
do we really need the `goto` here?   If this were just a standard `break`,  it'll naturally fall out of the case at this point and hit the bottom of the `while`, which bounces it right back up naturally.

Reply:
yeah, in this particular case it would work. But this is not how real e2e program would be structured. For example, there would be a way to get additional prompt from the user inside the loop, but then tool call results cannot go to the top of the while loop and ask the user for a new prompt. 

I noticed that in the mcp sample I did not have the loop at all, and then it kind of makes more sense. Alternatively I could turn this whole loop into a more complete program that asks the user for input (Console.ReadLine). Do you prefer any of these ways?

Reply:
If the "more complete, collect input" approach doesn't add a ton of complexity and make it harder to follow, that sounds great.   If not, then maybe eliding the loop altogether would be a better demonstration?

Reply:
OK. I will eliminate the loop. the interaction with the console/user kind of obscures the main points of the sample